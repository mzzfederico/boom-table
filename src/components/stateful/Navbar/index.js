import React, { useState } from "react";
import style from "styled-jsx/style";
import {withRouter, NavLink} from "react-router-dom";

const Navbar = () => {
    const [ openMenu, setOpenMenu ] = useState(false);

    return (
        <nav>
            <button className={!openMenu ? "menu-btn" : "menu-btn active"} onClick={() => setOpenMenu(!openMenu)}>menu</button>
            <ul className={!openMenu ? "closed" : ""}>
                <ListLink disableActive to={"/"}>Summary</ListLink>
                <ListLink to={"/clients"}>Clients</ListLink>
                <ListLink to={"/photoshoots"}>Photoshoots</ListLink>
                <ListLink to={"/typologies"}>Typologies</ListLink>
            </ul>

            <style jsx>{`
            nav {
                margin-bottom: 1rem;
            }
            ul {
                list-style: none;
                margin: 0;
                padding: 0;
            }
            .menu-btn {
                background: 0;
                border: 1px solid #aaa;
                margin-bottom: 0.5rem;
                padding: 0.25rem 0.5rem;
            }
            .menu-btn {
                background: 0;
                border: 1px solid #aaa;
                margin-bottom: 0.5rem;
                padding: 0.25rem 0.5rem;
            }
            .menu-btn.active {
                background: #aaa;
                color: white;
            }
            @media screen and (max-width: 640px) {
                ul.closed {
                    display: none;
                }
                li {
                    margin-bottom: 0.5rem;
                }
            }
            @media screen and (min-width: 641px) {
                .menu-btn {
                    display: none;
                }
                li {
                    display: inline;
                }
            }
            li:nth-child(1) {
                margin-right: 1rem;
            }
            li a {
                padding: 0.25rem;
                color: black;
                text-decoration: none;
            }
            li a.active {
                border-bottom: 2px solid black;
            }
        `}</style>
        </nav>
    );
};

const ListLink = ({ children, disableActive = false, ...linkProps }) => <li><NavLink activeClassName={!disableActive ? "active" : ""} {...linkProps}>{children}</NavLink></li>;

export default withRouter(Navbar);