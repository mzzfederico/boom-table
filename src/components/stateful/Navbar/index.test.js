/* eslint-disable no-undef */
import React from "react";
import Navbar from "./index";
import {MemoryRouter} from "react-router-dom";
import renderer from "react-test-renderer";

it("renders without crashing", () => {
    const tree = renderer.create(
        <MemoryRouter>
            <Navbar />
        </MemoryRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});