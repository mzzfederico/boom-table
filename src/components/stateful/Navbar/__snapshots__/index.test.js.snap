// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<nav>
  <ul>
    <li>
      <a
        aria-current="page"
        className=""
        href="/"
        onClick={[Function]}
        style={Object {}}
      >
        Summary
      </a>
    </li>
    <li>
      <a
        aria-current={null}
        href="/clients"
        onClick={[Function]}
      >
        Clients
      </a>
    </li>
    <li>
      <a
        aria-current={null}
        href="/photoshoots"
        onClick={[Function]}
      >
        Photoshoots
      </a>
    </li>
    <li>
      <a
        aria-current={null}
        href="/typologies"
        onClick={[Function]}
      >
        Typologies
      </a>
    </li>
  </ul>
  <style
    jsx={true}
  >
    
            nav {
                margin-bottom: 1rem;
            }
            ul {
                list-style: none;
                margin: 0;
                padding: 0;
            }
            li {
                display: inline;
            }
            li:nth-child(1) {
                margin-right: 1rem;
            }
            li a {
                padding: 0.25rem;
                color: black;
                text-decoration: none;
            }
            li a.active {
                border-bottom: 2px solid black;
            }
        
  </style>
</nav>
`;
